plugins/deployment_manager/DeploymentManager.ts

```diff
@@ -289,24 +356,26 @@ export class DeploymentManager {
     this.cache.writeCacheToDisk = writeCacheToDisk;
   }
 
-  /* Generates a new migration file, e.g. `deployments/<network>/<deployment>/migrations/1644385406_my_new_migration.ts`
-   **/
+  /* Generates a new migration file, e.g. `deployments/<network>/<deployment>/migrations/1644385406_my_new_migration.ts` */
   async generateMigration(name: string, timestamp?: number): Promise<string> {
-    return await generateMigration(this.cache, name, timestamp);
+    return generateMigration(this.cache, name, timestamp);
```

Comment:
FWIW, I still think `return await` is more expressive.

Reply:
I guess we weren't doing it totally consistently, so I just picked a style. But if we wanted to lint that everything should have the `await` I'm not really opposed

Reply:
Also there were some cases where it's actually important, so I was touching all the ones and making them consistent as I encountered the functions

Reply:
I guess I tend to feel like it erases type information, since it's not obvious that the return value is, in fact, a promise. E.g.

```ts
async f() {
  return g();
}
```

```ts
async f() {
  return await g();
}
```

It's more obvious to me that in the latter `g` is a async function or returns a promise, where the in the top one, it's not exceedingly obvious. As far as I know, these shouldn't be a case where the result is different, since awaits are synatic sugar and, I believe, should be erased when they end a function.

Reply:
Just looking into this, it looks like this will become (respecitvely): https://tc39.es/ecma262/#await

```ts
function f() {
  return Promise.resolve(g());
}
```

```ts
function f() {
  return Promise.resolve(g()).then((x) => {
    return Promise.resolve(x);
  }
}
```

These should be functionally equivalent, though the latter may be slightly slower depending on the quality of the JIT compiler.